apps/meteor/app/livechat/server/lib/LivechatTyped.ts

```diff
@@ -459,6 +460,31 @@ class LivechatClass {
 			extraData,
 		});
 
+		if (process.env.TEST_MODE?.toUpperCase() === 'TRUE') {
+			const visitorContact = await LivechatVisitors.findOne<Pick<ILivechatVisitor, 'contactId'>>(visitor._id, {
```

Comment:
In theory you shouldn't need to run a query here since this function already expects a full ILivechatVisitor, however I noticed two problems:
1. Calls from the apps-engine do not include the contactId, as the visitor converter is dropping this information (it needs to be fixed there)
2. The `findGuest` function that is used by some of the places that call this createRoom, is reading the visitor with a projection, loading only six attributes from the mongo document.

So I ask that you either:
1. Fix the places that call `createRoom` to send a full visitor object
2. Change the param type here to `AtLeast<ILivechatVisitor, ...>` specifying the attributes that are required by this function.

Reply:
I added the `contactId` to the visitor entity in some places that I could find that call the `createRoom`, but I probably missed some places. I kept a verification to ensure that we will have the contact id.